budget/__main__.py

```diff
@@ -70,19 +92,21 @@ def validate(transaction, categories):
 
 # parses transaction fields & inserts a date field (today) if not specified
 def parseTransaction(params):
+    dateAdded = False;
     transaction = [e.strip() for e in params.split(',')]
     if len(transaction) is NUM_TRANSACTION_FIELDS - 1:
         print("Only 3 fields were specified. Assigning today to date field.")
         #transaction.insert(0, str(datetime.now())[:10])
         transaction.insert(0, datetime.now())
+        dateAdded = True;
     if len(transaction) != NUM_TRANSACTION_FIELDS:
         raise UserWarning("Invalid number of fields in transaction.")
     try:
         if float(transaction[1]) <= 0 or float(transaction[1]) > 99999:
             raise ValueError()
     except ValueError:
             raise UserWarning("Invalid transaction amount: {0}".format(transaction[1]))
-    return transaction
+    return transaction, dateAdded
```

Comment:
Since this now returns 2 variables, you have to also update other places that this is called:
```py
# while processing the expense/income command
transaction, _ = parseTransaction(param)
```

Reply:
I am on it right away